From 22c42420e878f6eb3e3b6de83b37198f025ee8e0 Mon Sep 17 00:00:00 2001
From: tstegema <tstegema@90b65887-3827-0410-9a23-83215b262276>
Date: Thu, 9 Mar 2017 18:44:36 +0000
Subject: [PATCH 0827/1076] MANTIS 0045120 bugfix: RADAR_PRE_LEAK_32 kills
 mControl

trace.dll, tracehdl.dll, traceNoConfig.dll, tracehdlNoConfig.dll
V1.0.0.9
- when the command queue limit is reached the commands were ignored but not deleted



git-svn-id: svn://localhost/SelfServiceCommon/trunk@1169 90b65887-3827-0410-9a23-83215b262276
---
 SelfServiceCommon/Massai/cpp/Trace/inc/local_def.h    | 11 ++++++++---
 .../Massai/cpp/Trace/src/TraceCmdQueue.cpp            |  2 +-
 2 files changed, 9 insertions(+), 4 deletions(-)

diff --git a/SelfServiceCommon/Massai/cpp/Trace/inc/local_def.h b/SelfServiceCommon/Massai/cpp/Trace/inc/local_def.h
index 1b8e7e09..a22dc036 100644
--- a/SelfServiceCommon/Massai/cpp/Trace/inc/local_def.h
+++ b/SelfServiceCommon/Massai/cpp/Trace/inc/local_def.h
@@ -1,5 +1,5 @@
-#define FILE_VER_BIN     1,0,0,8
-#define FILE_VER_STR     "FileVersion", "1.0.0.8"
+#define FILE_VER_BIN     1,0,0,9
+#define FILE_VER_STR     "FileVersion", "1.0.0.9"
 
 /*!
 \file
@@ -9,10 +9,15 @@
 
 \section changehistory Change history
 
-\version 1.0.0.8
+\version 1.0.0.9
 
 trace.dll, tracehdl.dll, traceNoConfig.dll, tracehdlNoConfig.dll
 
+\version 1.0.0.9
+\li 09.03.2017
+\li bugfix SteT - Mantis 0045120: RADAR_PRE_LEAK_32 kills mControl
+- when the command queue limit is reached the commands were ignored but not deleted
+
 \version 1.0.0.8
 \li 01.03.2016
 \li bugfix SteT - Mantis 0037106: Trace.dll: call Config.removeListener on closeTrace
diff --git a/SelfServiceCommon/Massai/cpp/Trace/src/TraceCmdQueue.cpp b/SelfServiceCommon/Massai/cpp/Trace/src/TraceCmdQueue.cpp
index 1f885030..e3d21153 100644
--- a/SelfServiceCommon/Massai/cpp/Trace/src/TraceCmdQueue.cpp
+++ b/SelfServiceCommon/Massai/cpp/Trace/src/TraceCmdQueue.cpp
@@ -78,6 +78,7 @@ namespace MASSAI
           if(m_commandList.size() <= UPPER_BOUND)
           {
             m_commandList.push_back(pCmd);
+            send= true;
           }
           if(m_waitCount > 0)
           {
@@ -88,7 +89,6 @@ namespace MASSAI
             m_waitCount= WAIT_COUNT;
             wait= true;
           }
-          send= true;
           m_event.post();
           if(!m_started)
           {
-- 
2.41.0.windows.1

